src/main.rs

```diff
@@ -1,20 +1,98 @@
 pub mod engine;
 pub mod rpc;
+use kzg::G1;
+use kzg::G2;
+use std::io::Write;
+use tracing::info;
+
+use crate::engine::backend::Backend as _;
+use crate::engine::{backend::BackendConfig, blst::BlstBackend as Backend};
 
 use clap::Parser;
 
 #[derive(Parser, Debug)]
-#[command(name = "fourier-rpc", version = "0.1.0", about = "Fourier RPC server", long_about = None)]
-struct Args {
-    #[arg(short = 'H', long, default_value = "127.0.0.1")]
-    host: String,
-    #[arg(short = 'P', long, default_value = "1337")]
-    port: u16,
+#[clap(name = "fourier-rpc", version = "0.1.0", about = "Fourier RPC server", long_about = None)]
+struct CLi {
+    #[clap(flatten)]
+    global_opts: GlobalOpts,
+
+    #[clap(subcommand)]
+    subcmd: SubCommand,
+}
+
+#[derive(Parser, Debug)]
+pub struct GlobalOpts {}
+
+#[derive(Parser, Debug)]
+enum SubCommand {
+    Setup(SetupArgs),
```

Comment:
I think that principally, we don't want to expose this to users since this may cause confusion. Or maybe we should add some sort of print that tells users it will make them deviate from the subnet and therefore cause loss of emissions